lib/pwwka/receiver.rb

```diff
@@ -42,7 +42,11 @@ def self.subscribe(handler_klass, queue_name, routing_key: "#.#", block: true)
               log_error "Error Processing Message", error_options
               receiver.nack(delivery_info.delivery_tag)
             end
-            unless Pwwka.configuration.keep_alive_on_handler_klass_exceptions?
+            if handler_klass.respond_to?(:on_unhandled_error)
+              handler_klass.on_unhandled_error(e)
```

Comment:
can we pass pass in the args that came into `::handle!` here as well, so that `::on_unhandled_error` can call `::handle!` again with the same args after doing other work?

Reply:
We could, but by the time this has been called, the message has been `nack`ed and requeued, so `handle!` will get called again.

That said, if this happens a second time, you'd lose the message, so I think maybe this could be enhanced to be more resilient.  I'll look into that (but for your test, you should be able to use this version)